adr/adr-responses-api.md

```diff
@@ -0,0 +1,52 @@
+
+1. **Llama Stack Agents API:** An older API that provides an explicit, stateful `Agent` construct on the server. The server manages the agent's state and conversation history within a `session`. While this simplifies state management for the client, this API is not aligned with the broader AI ecosystem's direction.
+2. **Llama Stack Responses API:** A newer, stateless API designed for compatibility with the OpenAI Responses API format. This API accepts a list of tools and a chat history and returns a model-generated response, which may include one or more tool calls. It does not provide a server-side `Agent` or `session` construct, requiring the client to manage the state of the interaction.
+
+The strategic direction from the Llama Stack team is to prioritize the Responses API to maximize compatibility with the vast ecosystem of tools, SDKs, and developer knowledge built around the OpenAI API standard. The older Agents API could be deprecated, heavily refactored, or receive minimal future investment, posing a significant long-term risk.
+
+The core architectural question is: How can we build a robust, developer-friendly agentic framework on a stateless API without sacrificing the usability that an explicit `Agent` construct provides?
+
+## Decision
+
+We will adopt the **Llama Stack Responses API** as the exclusive backend interface for agentic reasoning within Red Hat's AI products.
+
+To address the absence of a server-side agent object, we will **design and implement a client-side Agent construct**. This client-side agent construct will be responsible for:
+
+* **State Management:** Maintaining the conversation history (user messages, AI responses, tool outputs).
+* **Agentic Loop:** Orchestrating the turn-by-turn interaction with the Llama Stack Responses API. This involves:
+    1. Formatting the current state (chat history, available tools) into a request.
+    2. Sending the request to the API endpoint.
+    3. Parsing the API response to distinguish between textual replies and tool call requests.
+    4. Executing any requested tool calls using client-side or remote tooling.
+    5. Formatting the tool execution results.
+    6. Appending the tool results to the history and repeating the loop until a final answer is generated or a stop condition is met.
+* **Tool Dispatching:** Managing the registration and invocation of available tools on the client side.
+
+The decision of what form this client-side Agent construct will take will be addressed in a subsequent ADR. Some proposals include adding the functionality to the official Llama Stack client library, creating a new SDK maintained by Red Hat (perhaps as part of OpenDataHub), providing a container image with a simple HTTP API, and adopting a pre-existing SDK such as LangChain. That ADR will cover the pros and cons of these options and propose a final resolution.
```

Comment:
```suggestion
The decision of what form this client-side Agent construct will take will be addressed in a subsequent [ADR](https://github.com/opendatahub-io/agents/pull/2/files).
```

Reply:
Hmm.  This seems a big extreme, but I am OK with it if there is no objection.  Let's see if anyone pushes back.